src/utils/vvim.cpp

```diff
@@ -1029,6 +1101,12 @@ bool VVim::handleKeyPressEvent(int key, int modifiers, int *p_autoIndentPos)
             }
 
             setMode(VimMode::Insert);
+        } else if (modifiers == Qt::ShiftModifier) {
+            // S, delete line and go to insert mode
+            tryGetRepeatToken(m_keys, m_tokens);
+            addActionToken(Action::Change);
```

Comment:
It is legal only when currently there is no Action token. Key press such as `dS` will crash here.